lib/Group_Proxy.php

```diff
@@ -117,10 +127,15 @@ public function inGroup($uid, $gid) {
 	public function getUserGroups($uid) {
 		$groups = array();
 
-		foreach($this->backends as $backend) {
-			$backendGroups = $backend->getUserGroups($uid);
-			if (is_array($backendGroups)) {
-				$groups = array_merge($groups, $backendGroups);
+		foreach($this->backends as $configPrefix => $backend) {
```

Comment:
> There is a risk here that a process calls getUserGroups() and sees a user suddendly isn't in a specific group any more

Within LDAP it shouldn't happen. If the servers have a different set of users and groups, it won't be possible that one user from one server is in a group of the other server. If a user isn't accesible, their group won't be neither.

Regarding custom groups, it's probably something that core should handle. The backend should be able to notify core that it can fetch the users or groups somehow, usually with the `ServerNotAvailableException` or something similar. However, if core doesn't handle this situation it doesn't matter what we do, there will be something causing issues.